### DISCUSSION TITLE
想了解一下 OSC 图标元素的引入问题

### DISCUSSION THREAD

**Discussion by techz1** [hooke007/mpv_PlayKit#134] [Q&A 问答] [answered]
MPC 系的有个 toolbar 图片可以替换主题，MPV 里有相应的规范么？我看到有个 morden 的 osc 主题，它的图标元素竟然是嵌在一个 Material Design Iconic 的字体里面...... 还有图标的位置是怎么换的，不是太麻烦的话，我想弄个类似网站播放器的 UI

**Comment by hooke007** [comment, marked as answer]
没有规范，因为mpv没有GUI
OSC本质是<s>字幕</s>（部分使用了ass的语法），它自带的图标也是用的字体里的（编译的时候生成的osd_font.otf）

弄成网页播放器那样的话，以bottombar为模板进行修改比较方便

**Reply by hooke007** [reply]
此外也可以看一下另一个osc作者的开发记录，他也是基于bottombar改的
https://github.com/Zren/mpv-osc-tethys/issues/4

**Comment by techz1** [comment]
自带的那几进度条有点不好看，我其实喜欢网页播放器那种细条，然后条在突变上方，尽量靠近底部, 以modern为底，有没有可能调成这种：
![2](https://user-images.githubusercontent.com/106645798/171401546-b060e722-2903-4987-92aa-c8c0506fddb1.jpg)

现在是这样

![1](https://user-images.githubusercontent.com/106645798/171401633-8fda2eb0-b0fa-4ddb-b5dd-a74b76bd0b51.jpg) 底部太高了

请教大佬大概是需要改哪里的哪些位置呢? 还有，B站那个 dash player 有没有可以抄的东西

**Reply by hooke007** [reply]
你说的这两都没没仔细看过（
内建的元素也照样能改啊
![Snipaste_2022-06-01_13-31-35](https://user-images.githubusercontent.com/41094733/171405113-64e414f7-916c-4b99-90d1-fe84161f74df.jpg)

**Comment by techz1** [comment]
我切换回 bottombar 照着文档试了几个参数不太理想，首先怎么把进度条的边框去掉，就像网页播放器一样留条细线　然后把它整个上移到按钮上方？　　
现在是这样：
![3](https://user-images.githubusercontent.com/106645798/171415154-fd3fee50-9f7f-4af4-a2f6-3712e1fb8fdc.jpg)

**Reply by hooke007** [reply]
。。它的代码没多复杂吧，geo里的y控制纵向位置。bgbar整段注释掉不就是去掉底色了吗
![Snipaste_2022-06-01_15-43-53](https://user-images.githubusercontent.com/41094733/171432396-c22e9065-4b9f-4f0a-a868-1b5cbbc4fc8d.jpg)

**Comment by techz1** [comment]
哦,我才看到上面那个回复的 tethys,谢谢。现在还有个基础性的问题，理论上，如果我只是单纯的修改UI,不加入其它它的例如视频索引，那是不是可以直接在原始的 osc上修改而不引入脚本呢?

**Reply by hooke007** [reply]
> 不加入其它它的例如视频索引

这个是什么意思。

原始的osc就是脚本，涉及外观选项自定义的部分基本都在脚本设置文件里了

**Reply by techz1** [reply]
嗯 我明白了谢谢指教。

索引那个，抱歉，我不知道那个实际叫什么，就是鼠标移过去的时出个缩略图，因为我看到一般带这个功能的 OSC 文件都单独列了个脚本

**Reply by hooke007** [reply]
。。。就叫缩略图
